### ISSUE TITLE
Jetty 12: stopping the app fails with `java.nio.file.ClosedFileSystemException`

### ISSUE THREAD

**Issue by mvysny** [mvysny/vaadin-boot#20]
Stopping the vaadin-boot app fails with:
```
2023-07-21 14:45:24.199 [main] INFO com.github.mvysny.vaadinboot.VaadinBoot - Main: Shutting down
2023-07-21 14:45:24.210 [main] INFO com.example.Bootstrap - Testapp shut down
2023-07-21 14:45:24.214 [main] ERROR com.github.mvysny.vaadinboot.VaadinBoot - stop() failed: java.nio.file.ClosedFileSystemException
java.nio.file.ClosedFileSystemException
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystem.ensureOpen(ZipFileSystem.java:1628)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystem.checkAccess(ZipFileSystem.java:560)
	at jdk.zipfs/jdk.nio.zipfs.ZipPath.checkAccess(ZipPath.java:892)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.checkAccess(ZipFileSystemProvider.java:172)
	at java.base/java.nio.file.Files.isAccessible(Files.java:2597)
	at java.base/java.nio.file.Files.isReadable(Files.java:2632)
	at org.eclipse.jetty.util.resource.PathResource.isReadable(PathResource.java:307)
	at org.eclipse.jetty.util.resource.Resources.isReadable(Resources.java:79)
	at org.eclipse.jetty.server.handler.ContextHandler.setBaseResource(ContextHandler.java:884)
	at org.eclipse.jetty.ee10.webapp.WebInfConfiguration.deconfigure(WebInfConfiguration.java:99)
	at org.eclipse.jetty.ee10.webapp.WebAppContext.stopContext(WebAppContext.java:1222)
	at org.eclipse.jetty.ee10.servlet.ServletContextHandler.doStop(ServletContextHandler.java:1070)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.stop(AbstractLifeCycle.java:132)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.stop(ContainerLifeCycle.java:182)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStop(ContainerLifeCycle.java:205)
	at org.eclipse.jetty.server.Handler$Abstract.doStop(Handler.java:403)
	at org.eclipse.jetty.server.Server.doStop(Server.java:648)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.stop(AbstractLifeCycle.java:132)
	at com.github.mvysny.vaadinboot.VaadinBoot.stop(VaadinBoot.java:366)
	at com.github.mvysny.vaadinboot.VaadinBoot.run(VaadinBoot.java:260)
	at com.example.Main.main(Main.java:7)
2023-07-21 14:45:24.215 [Thread-19] INFO com.github.mvysny.vaadinboot.VaadinBoot - Shutdown hook called, shutting down
2023-07-21 14:45:24.216 [Thread-19] ERROR com.github.mvysny.vaadinboot.VaadinBoot - stop() failed: java.lang.NullPointerException: Cannot invoke "javax.naming.Context.destroySubcontext(String)" because "compCtx" is null
java.lang.NullPointerException: Cannot invoke "javax.naming.Context.destroySubcontext(String)" because "compCtx" is null
	at org.eclipse.jetty.ee10.plus.webapp.EnvConfiguration.deconfigure(EnvConfiguration.java:165)
	at org.eclipse.jetty.ee10.webapp.WebAppContext.stopContext(WebAppContext.java:1222)
	at org.eclipse.jetty.ee10.servlet.ServletContextHandler.doStop(ServletContextHandler.java:1070)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.stop(AbstractLifeCycle.java:132)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.stop(ContainerLifeCycle.java:182)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStop(ContainerLifeCycle.java:205)
	at org.eclipse.jetty.server.Handler$Abstract.doStop(Handler.java:403)
	at org.eclipse.jetty.server.Server.doStop(Server.java:648)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.stop(AbstractLifeCycle.java:132)
	at com.github.mvysny.vaadinboot.VaadinBoot.stop(VaadinBoot.java:366)
	at com.github.mvysny.vaadinboot.VaadinBoot.lambda$run$0(VaadinBoot.java:246)
	at java.base/java.lang.Thread.run(Thread.java:833)
```
This only happens when the app is ran via the run script; this is not reproducible when main() is run from IDE.

Steps to reproduce:

1. ./gradlew
2. cd testapp/build/distributions/
3. unzip testapp-12.0-SNAPSHOT.zip
4. cd testapp/bin
5. ./testapp
6. When the app starts, press Enter to terminate the app

**Comment by mvysny** [comment]
Changing `VaadinBoot:330` to `Env.findWebRoot(ResourceFactory.root())` fixes the issue, however this creates another issue: the resources are no longer GCed when Jetty stops, they persist until the JVM itself stops.

I'll open a ticket at Jetty to ask for guidelines.

**Comment by mvysny** [comment]
Filed upstream: https://github.com/eclipse/jetty.project/issues/10134

**Comment by mvysny** [comment]
Upgraded Jetty to 12.0.0, the issue can no longer be reproduced :partying_face: 

Fixed by 06114789bd7507392f60dcd148817a05fa6a22a2
